torch/csrc/dynamo/guards.cpp

```diff
@@ -954,15 +962,16 @@ class EQUALS_MATCH : public LeafGuard {
 class TUPLE_ITERATOR_LEN : public LeafGuard {
  public:
   TUPLE_ITERATOR_LEN(
-      py::object length,
-      py::object type_id,
-      py::object verbose_code_parts)
+      const py::object& length,
+      const py::object& type_id,
+      const py::object& verbose_code_parts)
       : LeafGuard(verbose_code_parts),
         _length(py::cast<Py_ssize_t>(length)),
```

Comment:
Cast can take rvalue objects too

Reply:
But the specific cast takes a reference, as I verified by jumping into the definition.